### PR TITLE
feat: add plugin search functionality

### PR DESCRIPTION
**Pull request by walbercardoso**
## Summary

Added plugin search functionality. Useful to search multiples plugins

image

image

## Change Type

Please delete any irrelevant options.

- [x] New feature (non-breaking change which adds functionality)

## Testing

check if all the modified file works correctly
check the alignment and scaling of the icons

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: env/conda-meta/history [hunk 1]
```diff
@@ -0,0 +1,3 @@
+==> 2023-09-27 13:49:15 <==
+# cmd: /Users/walbercardoso/miniconda3/bin/conda create --prefix ./env
```
Comment (danny-avila) [inline env/conda-meta/history]: Can you delete this file from the PR?

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by danny-avila** [comment]
Thanks for the PR! Just one minor issue with the conda file

**Comment by danny-avila** [comment]
The Plugin Store tests need to be adjusted, too.

File: env/conda-meta/history (hunk 1 above)
Comment (walbercardoso) [reply env/conda-meta/history]: Done

**Comment by Berry-13** [comment]
There is a bug that, when it doesn't find anything, the plugin window becomes smaller.
I can't change the plugin page (only page 1 works).
The search bar doesn't adapt to the theme.

I'll try to fix it later

**Comment by walbercardoso** [comment]
Hi Berry. I just fixed. Can you try again?

**Comment by Berry-13** [comment]
@walbercardoso now the page work! do you mind if I make a commit that improves other things too?

**Comment by walbercardoso** [comment]
@Berry-13 Sure. Go ahead

**Comment by Berry-13** [comment]
OK I mess up with the commits, but the Dialog works perfectly. I've also added three translations and a new test for the search function. I think this PR is good to go. @walbercardoso thanks for your hard work and quick responses! Just let us know when you think it's ready

**Comment by walbercardoso** [comment]
Thanks @Berry-13 . I think its good to go now.
I'm happy to have contributed.

**Review by danny-avila — APPROVED** [review]

**Pull request merged**